.github/workflows/wasmcloud.yml

```diff
@@ -98,7 +98,47 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  meta:
+    runs-on: ubuntu-latest
+    outputs:
+      wasmcloud_modified: ${{ steps.changes.outputs.wasmcloud }}
+      providers_modified: ${{ steps.changes.outputs.providers }}
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+      - uses: dorny/paths-filter@v3
+        id: changes
+        with:
+          # Consider wasmCloud changed if any Rust file other than
+          # capability providers change
+          filters: |
+            wasmcloud:
+              - './.github/workflows/wasmcloud.yml'
+              - './.github/actions/build-nix/action.yml'
+              - './.github/actions/install-nix/action.yml'
+              - './.config/nextest.toml'
+              - 'nix/images/default.nix'
+              - 'flake.lock'
+              - 'flake.nix'
+              - 'Cargo.lock'
+              - 'Cargo.toml'
+              - 'rust-toolchain.toml'
+              - 'crates/**/*'
+              - 'crates/provider-sdk/*'
+              - 'src/**/*'
+              - 'tests/**/*'
+              - 'wit/**/*'
+              - '!crates/provider-*'
```

Comment:
Need to verify that the matching provider-sdk filter will override the not matching provider-* filter

Reply:
I think the ! will negate the change, but I will double check. Will consider a `!crates/provider-*-*` or otherwise structure